features/admin.server-configurations.v1/components/__tests__/password-recovery-form.test.tsx

```diff
@@ -0,0 +1,173 @@
+                "value": "5"
+            },
+            {
+                "description": "",
+                "displayName": "Notification based password recovery via an email",
+                "name": "Recovery.Notification.Password.emailLink.Enable",
+                "value": "false"
+            },
+            {
+                "description": "",
+                "displayName": "Notification based password recovery using SMS OTP",
+                "name": "Recovery.Notification.Password.smsOtp.Enable",
+                "value": "false"
+            }
+        ],
+        "subCategory": "DEFAULT"
+    },
+    isConnectorEnabled: true,
+    onSubmit: onSubmitMock
+};
+
+describe("PasswordRecoveryForm", () => {
+
+    it("renders the PasswordRecoveryForm component.", async () => {
+        render(<PasswordRecoveryConfigurationForm
+            { ...initialProps }
+            data-componentid="password-recovery-edit-form" />);
+
+        expect(screen.getByTestId(`${data_component_id}-email-link-based-recovery`)).toBeInTheDocument();
+    });
+
+    it("recovery option selection checkboxes are checkable.", async () => {
+        render(<PasswordRecoveryConfigurationForm { ...initialProps } />);
+
+        const email_checkbox: HTMLInputElement = screen.getByRole("checkbox", { name: "enableEmailBasedRecovery" });
+        const sms_checkbox: HTMLInputElement = screen.getByRole("checkbox", { name: "enableSMSBasedRecovery" });
+
+        fireEvent.click(email_checkbox);
+        fireEvent.click(sms_checkbox);
+
+        expect(email_checkbox).toBeChecked();
+        expect(sms_checkbox).toBeChecked();
+    });
+
+    it("otp code configuration checkboxes don't allow all checkboxes to be unchecked.", async () => {
+        render(<PasswordRecoveryConfigurationForm { ...initialProps } />);
+
+        const uppercase_checkbox: HTMLInputElement = screen.getByRole("checkbox",
+            { name: "passwordRecoveryOtpUseUppercase" });
+        const lowercase_checkbox: HTMLInputElement = screen.getByRole("checkbox",
+            { name: "passwordRecoveryOtpUseLowercase" });
+        const numeric_checkbox: HTMLInputElement = screen.getByRole("checkbox",
+            { name: "passwordRecoveryOtpUseNumeric" });
+
+        (!uppercase_checkbox.checked) && fireEvent.click(uppercase_checkbox);
+        (!lowercase_checkbox.checked) && fireEvent.click(lowercase_checkbox);
+        (numeric_checkbox.checked) && fireEvent.click(numeric_checkbox);
+        expect(uppercase_checkbox).toBeChecked();
+        expect(lowercase_checkbox).toBeChecked();
+        expect(numeric_checkbox).not.toBeChecked();
+        expect(numeric_checkbox).toBeDisabled();
+
+        (!uppercase_checkbox.checked) && fireEvent.click(uppercase_checkbox);
+        (!numeric_checkbox.checked) && fireEvent.click(numeric_checkbox);
+        (lowercase_checkbox.checked) && fireEvent.click(lowercase_checkbox);
+        expect(uppercase_checkbox).toBeChecked();
```

Comment:
Let's add unit tests to cover the following cases as well. This will help us catch any issues at the PR level instead of at the E2E level.

1. Whether the form values are sent to the intended endpoint on submitting the pwd recovery form.
2. Whether the input validations are working as intended.
3. Whether specific fields are getting enabled/disabled based on the state/prop values (if applicable)

Reply:
Ack